examples/hybrid/gravitywave_parameterization/gravity_wave_parameterization.jl

```diff
@@ -0,0 +1,228 @@
+
+    # compute buoyancy frequency
+    @. ᶜT = TD.air_temperature(thermo_params, ᶜts)
+
+    grad_scalar = similar(ᶜT)
+    parent(grad_scalar) .= parent(Geometry.WVector.(ᶜgradᵥ.(ᶠinterp.(ᶜT))))
+
+    ᶜbf = @. (grav / ᶜT) * (grad_scalar + grav / cp_m)
+    ᶜbf = @. ifelse(ᶜbf < FT(2.5e-5), FT(sqrt(2.5e-5)), sqrt(abs(ᶜbf))) # to avoid small numbers
+    # alternative
+    # ᶠbf = [i > 2.5e-5 ? sqrt(i) : sqrt(2.5e-5)  for i in ᶠbf]
+    # TODO: create an extra layer at model top so that the gravity wave forcing
+    # .     occurring between the topmost model level and the upper boundary
+    # .     may be calculated
+    #
+
+    # source level: get the index of the level that is closest to the source height (GFDL uses the fist level below instead)
+    source_level = argmin(
+        abs.(
+            unique(parent(Fields.coordinate_field(Y.c).z) .- gw_source_height)
+        ),
+    )
+    ᶠz = Fields.coordinate_field(Y.f).z
+
+    # TODO: prepare input variables for gravity_wave_forcing()
+    #       1. grab column wise data
+    #       2. convert coviant uv to physical uv
+    # need to make everything an array befere
+    u_phy = Geometry.UVVector.(Y.c.uₕ).components.data.:1
+    v_phy = Geometry.UVVector.(Y.c.uₕ).components.data.:2
+
+    uforcing = ones(axes(u_phy))
+    vforcing = similar(v_phy)
+    # bycolume
+    Fields.bycolumn(axes(ᶜρ)) do colidx
+        parent(uforcing[colidx]) .= gravity_wave_forcing(
+            parent(u_phy[colidx])[:, 1],
+            source_level,
+            gw_F_S0,
+            gw_Bm,
+            gw_c,
+            gw_cw,
+            gw_c0,
+            gw_nk,
+            gw_k,
+            gw_k2,
+            parent(ᶜbf[colidx])[:, 1],
+            parent(ᶜρ[colidx])[:, 1],
+            parent(ᶠz[colidx])[:, 1],
+        )
+        parent(vforcing[colidx]) .= gravity_wave_forcing(
+            parent(v_phy[colidx])[:, 1],
+            source_level,
+            gw_F_S0,
+            gw_Bm,
+            gw_c,
+            gw_cw,
+            gw_c0,
+            gw_nk,
+            gw_k,
+            gw_k2,
+            parent(ᶜbf[colidx])[:, 1],
+            parent(ᶜρ[colidx])[:, 1],
+            parent(ᶠz[colidx])[:, 1],
+        )
+
+    end
+
+    @. Yₜ.c.uₕ +=
+        Geometry.Covariant12Vector.(Geometry.UVVector.(uforcing, vforcing))
+
+end
+
+
+
+function gravity_wave_forcing(
+    ᶜu,
+    source_level,
+    F_S0,
+    Bm,
+    c,
+    cw,
+    c0,
+    nk,
+    kwv,
+    k2,
+    ᶜbf,
+    ᶜρ,
+    ᶠz,
+)
+
+    nc = length(c)
+
+    # define wave momentum flux (B0) at source level for each phase
+    # speed n, and the sum over all phase speeds (Bsum), which is needed
+    # to calculate the intermittency.
+
+    c_hat0 = c .- ᶜu[source_level]
+    # In GFDL code, flag is always 1: c = c0 * flag + c_hat0 * (1 - flag)
+    Bexp = @. exp(-log(2.0) * ( (c - c0) / cw)^2)
+    B0 = @. sign(c_hat0) * Bm * Bexp
+    # In GFDL code, it is: B0 = @. sign(c_hat0) * (Bw * Bexp + Bn * Bexp)
```

Comment:
Allocating a field, can we inline to avoid allocating or add to the cache to avoid allocating at every `step!`?